iam_mapping.py

```diff
@@ -46,7 +54,40 @@ def delete_mapping(body: dict, meta: dict, spec: dict, **kwargs):
     apply_mapping(cm, updated_mapping)
 
 
-def deploy_crd_definition():
+@kopf.on.startup()
+def on_startup(logger: logger, **kwargs) -> None:
+    # Do a full synchronization at the start
+    logger.info("Deploy CRD definition")
+    deploy_crd_definition()
+    logger.info("Reconcile all existing ressources")
+    full_synchronize()
+
+
+@kopf.on.probe(id='sync')
+def get_monitoring_status(**kwargs) -> str:
+    return check_synchronization()
+
+
+def check_synchronization() -> str:
+    cm = API.read_namespaced_config_map('aws-auth', 'kube-system')
+    identity_mappings = custom_objects_API.list_cluster_custom_object(
+        GROUP, VERSION, PLURAL)
+
+    users_in_cm = get_user_mapping(cm)
+    users_in_cm = users_in_cm if isinstance(users_in_cm, list) else list()
+    users_in_cm = [u['username'] for u in users_in_cm]
+
+    users_in_crd = [im["spec"]['username']
+                    for im in identity_mappings["items"]]
+
+    if (set(users_in_cm) == set(users_in_crd)):
+        return "in-sync"
```

Comment:
Not a big fan of returning a **str** as status code.
Either use a bool or an Enum (https://docs.python.org/3/library/enum.html)